docs/content/src/learn/guides/file-integrations/file-integration-using-ftp/README.md

```diff
@@ -0,0 +1,125 @@
+
+### Developing the FTP listener service
+
+Let's start implementation by importing the WSO2 FTP Connector in the `ftp_listener.bal` file which you just created. 
+This will pull the FTP Connector from Ballerina Central.
+
+```ballerina
+import wso2/ftp;
+```
+
+Next, let's create an FTP Listener instance by defining the configuration in the `Ballerina.conf` file. The `FTP_HOST` 
+is the IP address of the FTP server, while the `FTP_USERNAME` and `FTP_PASSWORD` are credentials of a user that has permission 
+to access the FTP server. The `FTP_HOST` is the port used to connect with the server, of which the default value is `21`.
+
+Then you can add the configurations for the type of files the listener should listen for. For instance, if listener 
+should be invoked for text files, the config for `FTP_FILE_NAME_PATTERN` should be set as `(.*).txt`. Next, add 
+the location to poll for files and how frequently the listener should poll for files, using the values 
+`FTP_LISTENER_PATH` and `FTP_POLLING_INTERVAL`respectively.
+
+<!-- INCLUDE_CODE_SEGMENT: { file: src/ftp_listener/ftp_listener.bal, segment: segment_1 } -->
+
+Create the service to listen to the FTP server using the above listener. When files are added or deleted on the server, 
+this service will be invoked, and the files will be processed.
+
+<!-- INCLUDE_CODE_SEGMENT: { file: src/ftp_listener/ftp_listener.bal, segment: segment_2 } -->
+
+Then implement the FTP Client, which will connect to the FTP server and get the details of new files to process. 
+     
+<!-- INCLUDE_CODE_SEGMENT: { file: src/ftp_listener/ftp_listener.bal, segment: segment_3 } -->
+
+Declare a map to store the details of processed files.
+
+```ballerina
+map<int> fileMap = {};
+```
+
+Now, implement the processing of added and deleted files. When files are added to the server, the FTP client will 
+retrieve the file size from the server, and the file name and its size will be added to the `fileMap`. When a file is 
+removed from the server, the file will be removed from the map.
+
+<!-- INCLUDE_CODE_SEGMENT: { file: src/ftp_listener/ftp_listener.bal, segment: segment_4 } -->
+
+## Testing
+
+### Invoking the service
+
+To begin with invoking the service, start the FTP server. 
+
+Navigate to `file-integration-using-ftp` directory and run the following command to build the listener service in `ftp_listener.bal`.
+
+```bash
+   $ ballerina build -a 
+```
+
+The successful build of a service will show us something similar to the following output.
+
+```
+Compiling source
+        wso2/ftp_listener:0.1.0
+
+Creating balos
+        target/balo/ftp_listener-2019r3-java8-0.1.0.balo
+```
+
+This will create the Ballerina executables inside the `/target` directory.
+
+Then run the jar file created in the above step.
+
+```bash
+   $ ballerina run ftp_listener --b7a.config.file=src/ftp_listener/resources/ballerina.conf 
```

Comment:
Shouldn't this be a java -jar command? The above text indicates a jar file.

Reply:
Updated.